### ISSUE TITLE
Zero training/validation errors but completely wrong in labeled images.

### ISSUE THREAD

**Issue by v738301** [spoonsso/dannce#130]
Hi !

I manually labeled 150 frames for com training. But after com-train, I found that the estimated COM in the validation images saved in debug_im_out_valid folder are all located at the left-top corner of the images. Which means the model didn't learn the task at all, however, the training error converged to zero rapidly in the training epochs.  Using the weights generated by com-train, I also found the predictions are all NaNs. (similar to #128 ). 

I have checked the video frames and labels before they enter the model.fit(), didn't spot anything wrong. 

Thank you so much!

- ### **Console outputs**

(dannce) PS C:\Users\jlab> cd C:\Users\jlab\Documents\MATLAB\dannce-master\demo\MyRat_1
(dannce) PS C:\Users\jlab\Documents\MATLAB\dannce-master\demo\MyRat_1> com-train ..\..\configs\com_rat_config.yaml
2022-11-07 18:00:02.093225: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cudart64_101.dll
io_config not found in io.yaml file, falling back to main config
batch_size not found in io.yaml file, falling back to main config
epochs not found in io.yaml file, falling back to main config
downfac not found in io.yaml file, falling back to main config
lr not found in io.yaml file, falling back to main config
num_validation_per_exp not found in io.yaml file, falling back to main config
debug not found in io.yaml file, falling back to main config
verbose not found in io.yaml file, falling back to main config
max_num_samples not found in io.yaml file, falling back to main config
crop_height not found in io.yaml file, falling back to main config
crop_width not found in io.yaml file, falling back to main config
com_train_dir set to: .\COM\train_results\
com_predict_dir set to: .\COM\predict_results\
dannce_train_dir set to: .\DANNCE\train_results\MAX\
dannce_predict_dir set to: .\DANNCE\predict_results\
n_views set to: 3
exp set to: [{'label3d_file': './20221107_171751_Label3D_com.mat'}]
io_config set to: io.yaml
batch_size set to: 2
epochs set to: 3
downfac set to: 2
lr set to: 5e-5
num_validation_per_exp set to: 10
debug set to: True
verbose set to: 1
max_num_samples set to: 100
crop_height set to: [0, 640]
crop_width set to: [0, 960]
base_config set to: ..\..\configs\com_rat_config.yaml
viddir set to: videos
camnames set to: None
n_channels_out set to: 1
sigma set to: 30
net set to: unet2d_fullbn
gpu_id set to: 0
immode set to: vid
mono set to: False
mirror set to: False
loss set to: mask_nan_keep_loss
num_train_per_exp set to: None
augment_hue set to: False
augment_brightness set to: False
augment_hue_val set to: 0.05
augment_bright_val set to: 0.05
augment_rotation_val set to: 5
data_split_seed set to: None
valid_exp set to: None
dsmode set to: nn
com_finetune_weights set to: None
augment_shift set to: False
augment_zoom set to: False
augment_shear set to: False
augment_rotation set to: False
augment_shear_val set to: 5
augment_zoom_val set to: 0.05
augment_shift_val set to: 0.05
start_batch set to: 0
n_channels_in set to: None
extension set to: None
vid_dir_flag set to: None
chunks set to: None
lockfirst set to: None
load_valid set to: None
drop_landmark set to: None
raw_im_h set to: None
raw_im_w set to: None
n_instances set to: 1
start_sample set to: 0
write_npy set to: None
use_npy set to: False
com_predict_weights set to: None
com_debug set to: None
com_exp set to: None
Using the following *dannce.mat files: .\20221107_131602_Label3D_dannce.mat
Setting vid_dir_flag to True.
Setting extension to .mp4.
Setting chunks to {'Camera1': array([0]), 'Camera2': array([0]), 'Camera3': array([0])}.
Setting n_channels_in to 3.
Setting raw_im_h to 640.
Setting raw_im_w to 960.
Experiment 0 using videos in .\videos
Experiment 0 using camnames: ['Camera1', 'Camera2', 'Camera3']
{'0_Camera1': array([0]), '0_Camera2': array([0]), '0_Camera3': array([0])}
./20221107_171751_Label3D_com.mat
Using nn downsampling
TRAIN EXPTS: [0]
Initializing Network...
2022-11-07 18:00:03.728649: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library nvcuda.dll
2022-11-07 18:00:03.748705: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties:
pciBusID: 0000:01:00.0 name: NVIDIA GeForce RTX 3070 computeCapability: 8.6
coreClock: 1.755GHz coreCount: 46 deviceMemorySize: 8.00GiB deviceMemoryBandwidth: 417.29GiB/s
2022-11-07 18:00:03.748763: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cudart64_101.dll
2022-11-07 18:00:03.748917: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cublas64_10.dll
2022-11-07 18:00:03.749027: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cufft64_10.dll
2022-11-07 18:00:03.749163: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library curand64_10.dll
2022-11-07 18:00:03.749258: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cusolver64_10.dll
2022-11-07 18:00:03.749352: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cusparse64_10.dll
2022-11-07 18:00:03.749445: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cudnn64_7.dll
2022-11-07 18:00:03.749575: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1858] Adding visible gpu devices: 0
2022-11-07 18:00:03.749825: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN)to use the following CPU instructions in performance-critical operations:  AVX2
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-11-07 18:00:03.754879: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x16ebdc47a10 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2022-11-07 18:00:03.754930: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
2022-11-07 18:00:03.755206: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1716] Found device 0 with properties:
pciBusID: 0000:01:00.0 name: NVIDIA GeForce RTX 3070 computeCapability: 8.6
coreClock: 1.755GHz coreCount: 46 deviceMemorySize: 8.00GiB deviceMemoryBandwidth: 417.29GiB/s
2022-11-07 18:00:03.755289: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cudart64_101.dll
2022-11-07 18:00:03.755429: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cublas64_10.dll
2022-11-07 18:00:03.755558: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cufft64_10.dll
2022-11-07 18:00:03.755658: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library curand64_10.dll
2022-11-07 18:00:03.755794: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cusolver64_10.dll
2022-11-07 18:00:03.755893: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cusparse64_10.dll
2022-11-07 18:00:03.756025: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cudnn64_7.dll
2022-11-07 18:00:03.756144: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1858] Adding visible gpu devices: 0
2022-11-07 18:01:55.707808: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1257] Device interconnect StreamExecutor with strength 1 edge matrix:
2022-11-07 18:01:55.707875: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1263]      0
2022-11-07 18:01:55.707914: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1276] 0:   N
2022-11-07 18:01:55.708200: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1402] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 6584 MB memory) -> physical GPU (device: 0, name: NVIDIA GeForce RTX 3070, pci bus id: 0000:01:00.0, compute capability: 8.6)
2022-11-07 18:01:55.710122: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x16ed9547d60 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2022-11-07 18:01:55.710164: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): NVIDIA GeForce RTX 3070, Compute Capability 8.6
COMPLETE

2022-11-07 18:01:56.348486: I tensorflow/core/profiler/lib/profiler_session.cc:164] Profiler session started.
2022-11-07 18:01:56.348558: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1391] Profiler found 1 GPUs
2022-11-07 18:01:56.349393: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'cupti64_101.dll'; dlerror: cupti64_101.dll not found
2022-11-07 18:01:56.349946: W tensorflow/stream_executor/platform/default/dso_loader.cc:59] Could not load dynamic library 'cupti.dll'; dlerror: cupti.dll not found
2022-11-07 18:01:56.349995: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1441] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error CUPTI could not be loaded or symbol could not be found.
Loading data
Loading new video: .\videos\Camera1\0.mp4 for 0_Camera1
Loading new video: .\videos\Camera2\0.mp4 for 0_Camera2
Loading new video: .\videos\Camera3\0.mp4 for 0_Camera3
Loading new video: .\videos\Camera1\0.mp4 for 0_Camera1
Loading new video: .\videos\Camera2\0.mp4 for 0_Camera2
Loading new video: .\videos\Camera3\0.mp4 for 0_Camera3
Saving debug images to: .\COM\train_results\debug_im_out
Epoch 1/3
2022-11-07 18:02:56.310381: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cudnn64_7.dll
2022-11-07 18:05:07.732174: W tensorflow/stream_executor/gpu/redzone_allocator.cc:314] Internal: Invoking GPU asm compilation is supported on Cuda non-Windows platforms only
Relying on driver to perform ptx compilation.
Modify $PATH to customize ptxas location.
This message will be only logged once.
2022-11-07 18:05:07.981978: I tensorflow/stream_executor/platform/default/dso_loader.cc:48] Successfully opened dynamic library cublas64_10.dll
 1/70 [..............................] - ETA: 0s - loss: 0.24542022-11-07 18:05:22.302753: I tensorflow/core/profiler/lib/profiler_session.cc:164] Profiler session started.
2022-11-07 18:05:22.302850: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1441] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error CUPTI could not be loaded or symbol could not be found.
WARNING:tensorflow:From C:\Users\jlab\anaconda3\envs\dannce\lib\site-packages\tensorflow\python\ops\summary_ops_v2.py:1277: stop (from tensorflow.python.eager.profiler) is deprecated and will be removed after 2020-07-01.
Instructions for updating:
use `tf.profiler.experimental.stop` instead.
2022-11-07 18:05:22.434877: I tensorflow/core/profiler/internal/gpu/device_tracer.cc:223]  GpuTracer has collected 0 callback api events and 0 activity events.
2022-11-07 18:05:22.438600: E tensorflow/core/profiler/utils/hardware_type_utils.cc:61] Invalid GPU compute capability.
70/70 [==============================] - ETA: 0s - loss: 0.0105WARNING:tensorflow:Callbacks method `on_test_batch_end` is slow compared to the batch time (batch time: 0.0000s vs `on_test_batch_end` time: 0.0156s). Check your callbacks.
70/70 [==============================] - 10s 137ms/step - loss: 0.0105 - val_loss: 0.0000e+00
Epoch 2/3
70/70 [==============================] - 9s 125ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Epoch 3/3
70/70 [==============================] - 9s 126ms/step - loss: 0.0000e+00 - val_loss: 0.0000e+00
Saving debug images to: .\COM\train_results\debug_im_out_valid
Renaming weights file with best epoch description
Saving full model at end of training

- ### **The training label saved in debug_im_out folder**

![Capture](https://user-images.githubusercontent.com/54001145/200275224-b6d576d0-9014-4139-b5ae-609624f70e5a.PNG)

- ### **The validation results saved in debug_im_out_valid folder**

![Capture](https://user-images.githubusercontent.com/54001145/200274258-dddc12ce-e09e-4432-8489-25af7f720954.PNG)

**Comment by v738301** [comment]
I just found that it is because my cudnn version is not compatible with my GPU, after I reinstalled dannce with correct cudnn version, the problem solved.

**Comment by histun** [comment]
Hi v738301.

I'm having a similar issue with com-predict. With the given demo config files and the markeless_mouse1 videos, it generates a com3d0.mat, which has 99 NaN values out of 100 predictions using the given weights: ./COM/train_results/weights.250-0.00036.hdf5. 

I've upgraded cudnn from 7.6.5 to 8.0.5.39, but it's still the same. I'm using rtx4090. Does anyone have any idea how to solve this?
